### PR TITLE
Created calcLinearTresspassScore function for rectangles

### LINKED ISSUE #526: create "calcLinearTrespassScore" utility function for rectangles
### Description of the task

Create a trespass function that indicates the trespass score of a point with respect to a rectangle.

### Outline the task in more detail

Trespass scores indicate how far into a "bad region" a given point is. The higher the score, the more the point is "trespassing" in the region.

This function should return a value in the range [0,1] that indicates how much the given point is trespassing in the region given by the rectangle. The function signature should be something like the following:
`unsigned int calcLinearTrespassScore(Rectangle r, Point p);`

The function should return a trespass score of 0 if `p` is outside the rectangle, and values that linearly increase to 1 when the point is exactly in the middle of the rectangle. When calculating how far inside the rectangle the point is, use the largest of the distances to the 2 closest edges.

The `linear` function from #522  can be used for the implementation

This function should go in the `src/thunderbots/software/ai/navigator/trespass.*` files.

### Acceptance criteria

- [ ] Implement the function
- [ ] unit test

### PR DESCRIPTION
**Pull request by roark-z**
### Description

Added calcLinearTresspassScore function for rectangles in `src/thunderbots/software/ai/navigator/trespass.*`

### Testing Done

Created unit tests for the function

### Resolved Issues

resolves #526 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/ai/navigator/trespass.cpp [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+            else
+            {
+                double width_w  = p.x() - r.seCorner().x();
+                double width_e  = r.nwCorner().x() - p.x();
+                double height_s = p.y() - r.seCorner().y();
+                double height_n = r.nwCorner().y() - p.y();
+                return fmax(Util::linear(fmin(width_w, width_e), r.width() / 4.0,
+                                         r.width() / 2.0),
+                            Util::linear(fmin(height_s, height_n), r.height() / 4.0,
+                                         r.height() / 2.0));
```
Comment (EvanMorcom) [inline src/thunderbots/software/ai/navigator/trespass.cpp]: I'd like to see some comments in this section; not the easiest to follow

File: src/thunderbots/software/test/ai/navigator/trespass/trespass.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Test file for trespass functions
+ */
+
+#include "ai/navigator/trespass.h"
+
+#include <gtest/gtest.h>
+
+#include "geom/point.h"
+#include "geom/rectangle.h"
+
```
Comment (EvanMorcom) [inline src/thunderbots/software/test/ai/navigator/trespass/trespass.cpp:11]: I know these are WIP, but please add a brief comment explaining what's being tested before each unit test

**Review by MathewMacDougall — COMMENTED** [review]
Is this still WIP? If not, make sure to change the PR title before merging. I have no additional comments except for @EvanMorcom's comments from before.

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**